### PR TITLE
Invert SBUS on RC serial connections on F4

### PR DESCRIPTION
**Pull request by andyp1per**
Confirmed that this fixes SBUS on KakuteF4Mini:
https://discuss.ardupilot.org/t/futaba-r7008sb-not-working-on-kakute-f4-v2-mini/74872/26

Betaflight only uses active high, so other definitions were either missing or wrong.

### CODE REVIEW

**Comment by Hwurzburg** [comment]
what about FPort , non telem connections? can this allow them to be used without an inverter on F4 also?

**Comment by andyp1per** [comment]
> currently the wiki warns that Sbus and Fport need external inverters on F4s when using the serial port RC protocol....this eliminates that correct?

Correct on boards that have a HW inverter. As far as I can tell from the betaflight pinouts the matek board doesn't. The ones that do support it that we support are Diatone, Omnibus, Holybro, Revo, Sparky, iFlight, Airbot

**Comment by andyp1per** [comment]
It might help Fport if the inverter is on the TX as well

File: libraries/AP_HAL_ChibiOS/UARTDriver.cpp [hunk 1]
```diff
@@ -1645,22 +1645,36 @@ bool UARTDriver::set_options(uint16_t options)
...
     }
-#else // STM32F4
+#elif defined(STM32F4)
     // F4 can do inversion by GPIO if enabled in hwdef.dat, using
     // TXINV and RXINV options
     if (options & OPTION_RXINV) {
         if (sdef.rxinv_gpio >= 0) {
             hal.gpio->write(sdef.rxinv_gpio, sdef.rxinv_polarity);
+            if (arx_line != 0) {
+                palLineSetPushPull(arx_line, PAL_PUSHPULL_PULLDOWN);
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/UARTDriver.cpp:1655]: likely don't want this as inverter changes idle state as well as active state

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/UARTDriver.cpp (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/UARTDriver.cpp]: Fixed

**Review by tridge — APPROVED** [review]

**Comment by andyp1per** [comment]
@rmackay9 I think this should go into 4.1

**Comment by rmackay9** [comment]
@andyp1per ok thanks.  I've added this to the 4.1 projects for Copter, Plane, Rover.

**Pull request merged**